src/components/Class/ClassForm.js

```diff
@@ -0,0 +1,720 @@
+    []
+  );
+  const [showSuccessModal, setShowSuccessModal] = useState(false);
+
+  //getting pathway courses
+  const dispatch = useDispatch();
+  const data = useSelector((state) => {
+    return state;
+  });
+
+  console.log("classFields", classFields);
+  console.log("classToEdit", classToEdit);
+
+  useEffect(() => {
+    dispatch(pathwayActions.getPathwaysCourse({ pathwayId: 1 }));
+  }, [dispatch, 1]);
+
+  const courses =
+    data.Pathways.data &&
+    data.Pathways.data.pathways[0] &&
+    data.Pathways.data.pathways[0].courses.map((item) => {
+      return {
+        label: item.name,
+        value: item.id,
+      };
+    });
+
+  const selectedCourseLabel = courses.find(
+    (item) => item.value === classFields.course_id
+  );
+
+  const selectedExerciseLabel = exercisesForSelectedCourse.find(
+    (item) => item.id === classFields.exercise_id
+  );
+
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+  const classes = useStyles();
+  const capEnrollment = ["No Limit", 10, 20, 30];
+  const days = {
+    MO: "Mon",
+    TU: "Tue",
+    WE: "Wed",
+    TH: "Thu",
+    FR: "Fri",
+    SA: "Sat",
+    SU: "Sun",
+  };
+
+  const changeHandler = (e) => {
+    setClassFields({ ...classFields, [e.target.name]: e.target.value });
+  };
+
+  const handleDaySelection = (e) => {
+    const index = classFields.on_days.indexOf(e.target.value);
+    if (index === -1) {
+      setClassFields({
+        ...classFields,
+        ["on_days"]: [...classFields.on_days, e.target.value],
+      });
+    } else {
+      const dayDeleted = classFields.on_days.filter(
+        (selectedDay) => selectedDay !== e.target.value
+      );
+      setClassFields({ ...classFields, ["on_days"]: dayDeleted });
+    }
+  };
+
+  useEffect(() => {
+    console.log("id", classFields.partner_id);
+  }, [classFields.partner_id]);
+
+  useEffect(() => {
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/partners`,
+      headers: {
+        accept: "application/json",
+        "version-code": versionCode,
+        Authorization: user.data.token,
+      },
+    }).then((res) => {
+      // console.log("res", res);
+      const partners = res.data.partners.map((item, index) => {
+        return {
+          label: item.name,
+          id: item.id,
+        };
+      });
+      setPartnerData(partners);
+    });
+  }, []);
+
+  const convertToIST = (d) => {
+    const b = d.split(/\D+/);
+    const dateInObj = new Date(
+      Date.UTC(b[0], --b[1], b[2], b[3], b[4], b[5], b[6])
+    );
+    const utc = dateInObj.getTime() + dateInObj.getTimezoneOffset() * 60000;
+    return new Date(utc + 3600000 * +5.5).toISOString();
+  };
+
+  const createClass = (payload) => {
+    payload.start_time = convertToIST(payload.start_time);
+    payload.end_time = convertToIST(payload.end_time);
+    return axios({
+      url: `${process.env.REACT_APP_MERAKI_URL}classes`,
```

Comment:
Remember that we'll need a slash in the path to address https://github.com/navgurukul/bhanwari-devi/pull/431#discussion_r891339459 :

> IMPORTANT: There are a number of places that do not contain a trailing / after ${process.env.REACT_APP_MERAKI_URL}. From what I can tell, this will refer to https://dev-api.merakilearn.org/ in the development environment, but https://api.merakilearn.org (without the trailing /) in production. So, this would refer to say https://api.merakilearn.orgpathways/1/upcomingBatches, causing bad requests and lots of problems after merging into main (You can see these requests if you open up the Developer console and check the Network tab before visiting https://pr-431.dkchei85ij0cu.amplifyapp.com/course-content/1/370/1).
```suggestion
      url: `${process.env.REACT_APP_MERAKI_URL}/classes`,
```

Reply:
Ohh, yes... I had forgot, thanks.